### ISSUE TITLE
Interface only does not handle __call__

### ISSUE THREAD

**Issue by tim-mitchell** [seequent/pure_interface#83]
Apparently `__getattr__` is not called to find the `__call__` attribute when calling an object.

```
def f():
    print('f called')
    return True

class A:
    def __getattr__(self, item):
        if item == '__call__':
            return f

a = A()
v = a()
TypeError: 'A' object is not callable
```
This means that `Interface.adapt(obj)` breaks when the interface includes `__call__`

**Comment by mpj17** [comment]
You want [`__getattribute__`.](https://docs.python.org/3/reference/datamodel.html#object.__getattribute__)

**Comment by tim-mitchell** [comment]
@mpj17 Nope - that doesn't work either.
Strangely other double underscore methods, such as `__len__` do work in this scenario.

**Comment by tim-mitchell** [comment]
Fixed in 5.0.1
